provider/cluster_autoscaler_resource.go

```diff
@@ -0,0 +1,703 @@
+			fmt.Sprintf(
+				"Failed getting autoscaler for cluster '%s': %v",
+				plan.Cluster.Value, err,
+			),
+		)
+		return
+	}
+
+	autoscaler, err := clusterAutoscalerStateToObject(plan)
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Failed updating cluster autoscaler",
+			fmt.Sprintf(
+				"Failed updating autoscaler for cluster '%s: %v ",
+				plan.Cluster.Value, err,
+			),
+		)
+		return
+	}
+
+	update, err := r.collection.Cluster(plan.Cluster.Value).
+		Autoscaler().Update().Body(autoscaler).SendContext(ctx)
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Failed updating cluster autoscaler",
+			fmt.Sprintf(
+				"Failed updating autoscaler for cluster '%s': %v",
+				plan.Cluster.Value, err,
+			),
+		)
+		return
+	}
+
+	object := update.Body()
+	state := &ClusterAutoscalerState{}
+	populateAutoscalerState(object, plan.Cluster.Value, state)
+
+	diags = response.State.Set(ctx, state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (r *ClusterAutoscalerResource) Delete(ctx context.Context, request tfsdk.DeleteResourceRequest,
+	response *tfsdk.DeleteResourceResponse) {
+	state := &ClusterAutoscalerState{}
+	diags := request.State.Get(ctx, state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	resource := r.collection.Cluster(state.Cluster.Value).Autoscaler()
+	_, err := resource.Delete().SendContext(ctx)
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Failed deleting cluster autoscaler",
+			fmt.Sprintf(
+				"Failed deleting autoscaler for cluster '%s': %v",
+				state.Cluster.Value, err,
+			),
+		)
+		return
+	}
+
+	response.State.RemoveResource(ctx)
+}
+
+func (r *ClusterAutoscalerResource) ImportState(ctx context.Context, request tfsdk.ImportResourceStateRequest,
+	response *tfsdk.ImportResourceStateResponse) {
+	clusterId := request.ID
+
+	getResponse, err := r.collection.Cluster(clusterId).Autoscaler().Get().SendContext(ctx)
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Failed importing cluster autoscaler",
+			err.Error(),
+		)
+		return
+	}
+
+	state := &ClusterAutoscalerState{}
+	populateAutoscalerState(getResponse.Body(), clusterId, state)
+	diags := response.State.Set(ctx, state)
+	response.Diagnostics.Append(diags...)
+}
+
+func (r *ClusterAutoscalerResource) waitForClusterToGetReady(ctx context.Context, clusterId string) error {
+	resource := r.collection.Cluster(clusterId)
+
+	pollCtx, cancel := context.WithTimeout(ctx, 1*time.Hour)
+	defer cancel()
+
+	_, err := resource.Poll().
+		Interval(30 * time.Second).
+		Predicate(func(get *cmv1.ClusterGetResponse) bool {
+			return get.Body().State() == cmv1.ClusterStateReady
+		}).
+		StartContext(pollCtx)
+
+	return err
+}
+
+// populateAutoscalerState copies the data from the API object to the Terraform state.
+func populateAutoscalerState(object *cmv1.ClusterAutoscaler, clusterId string, state *ClusterAutoscalerState) {
```

Comment:
Basically speaking OCM doesn't come up with values that weren't set by the client. Meaning there are no defaults or something alike. In this function we take an API object and set the state based on it

Should it still be changed to ``comptued = true``? Because I think the usage for it is a bit different (IDs being generated, defaults being set by the backend, etc.)

Reply:
We already had similar issue in which the backend was changed and added default value, any future change like ths might cause to break the terraform, so we must take care for it also here.
If there is no chance for updating the value, so just lets remove it from the populate, otherwise need to declare it can be computed